src/main/java/ch/njol/skript/registrations/EventValues.java

```diff
@@ -135,203 +148,224 @@ public String getExcludeErrorMessage() {
 	 * 
-	 * @param e the event class the getter will be getting from
-	 * @param c type of getter
-	 * @param time the event-value's time
-	 * @return A getter to get values for a given type of events
-	 * @see #registerEventValue(Class, Class, Getter, int)
-	 * @see EventValueExpression#EventValueExpression(Class)
+	 * @param event the event class the getter will be getting from.
+	 * @param c type of getter.
+	 * @param time the event-value's time.
+	 * @return A getter to get values for a given type of events.
+	 * {@link #registerEventValue(Class, Class, Getter, int)}
+	 * {@link EventValueExpression#EventValueExpression(Class)}
+	 * @deprecated Due for removal, use {@link EventValues#getEventValueConverter(Class, Class, int)}
 	 */
+	@Deprecated
 	@Nullable
-	public static <T, E extends Event> Getter<? extends T, ? super E> getEventValueGetter(Class<E> e, Class<T> c, int time) {
-		return getEventValueGetter(e, c, time, true);
+	// Note: Skript doesn't use this method anywhere. Purely external addons deprecation.
+	public static <T, E extends Event> Getter<? extends T, ? super E> getEventValueGetter(Class<E> event, Class<T> c, int time) {
+		Converter<E, T> converter = getEventValueConverter(event, c, time, true);
+		return new Getter<T, E>() {
+			@Override
+			public @Nullable T get(E event) {
+				return converter.convert(event);
+			}
+		};
 	}
-	
+
+	/**
+	 * Returns a converter to get a value from in an event.
+	 * <p>
+	 * Can print an error if the event value is blocked for the given event.
+	 * 
+	 * @param event the event class the getter will be getting from.
+	 * @param c type of converter.
+	 * @param time the event-value's time.
+	 * @return A getter to get values for a given type of events.
+	 * {@link #registerEventValue(Class, Class, Getter, int)}
+	 * {@link EventValueExpression#EventValueExpression(Class)}
+	 */
+	@Nullable
+	public static <T, E extends Event> Converter<E, T> getEventValueConverter(Class<E> event, Class<T> c, int time) {
+		return getEventValueConverter(event, c, time, true);
+	}
+
 	@SuppressWarnings("unchecked")
 	@Nullable
-	private static <T, E extends Event> Getter<? extends T, ? super E> getEventValueGetter(Class<E> e, Class<T> c, int time, boolean allowDefault) {
+	private static <T, E extends Event> Converter<E, T> getEventValueConverter(Class<E> event, Class<T> c, int time, boolean allowDefault) {
 		List<EventValueInfo<?, ?>> eventValues = getEventValuesList(time);
+
 		// First check for exact classes matching the parameters.
-		for (EventValueInfo<?, ?> ev : eventValues) {
-			if (!c.equals(ev.c))
+		for (EventValueInfo<?, ?> info : eventValues) {
+			if (!c.equals(info.getValueClass()))
 				continue;
-			if (!checkExcludes(ev, e))
+			if (!checkExcludes(info, event))
 				return null;
-			if (ev.event.isAssignableFrom(e))
-				return (Getter<? extends T, ? super E>) ev.getter;
+			if (info.getEventClass().isAssignableFrom(event))
+				return (Converter<E, T>) info.getConverter();
```

Comment:
This isn't a valid cast, the converter is a `Converter<? super E, T>` which cannot be cast to a `Converter<E, T>`. Example:
![image](https://user-images.githubusercontent.com/29547183/184876398-888cda0e-2bcf-436f-852a-bfe2f4304755.png)

You'll probably need to change the signatures back to what they previously were

Reply:
cannot replicate

Reply:
What do you mean, can you send your code?

Reply:
If it didn't work Skript tests wouldn't be passing.

Reply:
Java doesn't have checked casts for types with generics, so they don't throw exceptions, for example:
![image](https://user-images.githubusercontent.com/29547183/185081508-fe4472b2-9d9b-4842-92d3-dd1d78ffb826.png)

This is what the `unchecked` warning suppressor is for.

Reply:
Yes, but it's the correct cast. It's all E extends Event

Reply:
It's not correct, in generic types a class cannot be converted to its superclass:
![image](https://user-images.githubusercontent.com/29547183/185088369-dd8ccb7a-5ed6-43ea-ae81-58f277388ac8.png)

Reply:
Unresolved FYI